synAnnotationUtils/update.py

```diff
@@ -64,154 +65,301 @@ def updateAnnoByIdDictFromDict(syn,idDict,annoDict,forceVersion = False):
             logging.info("Accessing File %s ..." % synEntity.id)
-            _helperUpdateFormatTypeByFileName(syn,synEntity,annoKey,annoDict,forceVersion)
+            _helperUpdateFormatTypeByFileName(syn, synEntity, annoKey, annoDict, forceVersion)
     else:
         logging.info("Input is a Synpase ID")
-        synEntity = syn.get(synId,downloadFile = False)
+        synEntity = syn.get(synId, downloadFile=False)
         if not is_container(synEntity):
             logging.info("%s is a File" % synId)
-            _helperUpdateFormatTypeByFileName(syn,synEntity,annoKey,annoDict,forceVersion)
+            _helperUpdateFormatTypeByFileName(syn, synEntity, annoKey, annoDict, forceVersion)
         else:
-            directory = synapseutils.walk(syn,synId)
-            for dirpath,dirname,filename in directory:
+            directory = synapseutils.walk(syn, synId)
+            for dirpath, dirname, filename in directory:
                 for i in filename:
-                    synEntity = syn.get(i[1],downloadFile = False)
+                    synEntity = syn.get(i[1], downloadFile=False)
                     logging.info("Getting File %s ..." % i[1])
-                    _helperUpdateFormatTypeByFileName(syn,synEntity,annoKey,annoDict,forceVersion)
+                    _helperUpdateFormatTypeByFileName(syn, synEntity, annoKey, annoDict, forceVersion)
 
-def _helperUpdateFormatTypeByFileName(syn,synEntity,annoKey,annoDict,forceVersion):
-    logging.info("Updating %s..." % annoKey)
-    synEntityName = synEntity.name
-    synEntityType = ""
-    for ext in annoDict.keys():
-        if synEntityName.endswith(ext):
-            synEntityType = annoDict[ext]
-            synEntity[annoKey] = synEntityType
-            synEntity = syn.store(synEntity,forceVersion = forceVersion)
-            logging.info("Done!")
-            break
 
-    if synEntityType == "":
-        logging.warning("ERROR: File type not found in file types dictionary")
-    logging.info("")
+def _csv2df(path, indexCol):
+    """
+    Reads users .csv file containing updated annotations and converts it into a pandas dataframe
+    while removing all NA rows
+
+    :param path:      Current working directory absolute/relative path to user-defined manifest .csv file containing
+                      updated cells with the same schema as existing entity-view
+    :param indexCol:  Column or field name of the synapse ROW_IDs. Pandas library reads the first unnamed column as
+                      'Unnamed: 0'.
+    :return:          A dataframe containing modified information on an entity-view.
+                      Structured as synapse file Ids X standard synapse schema + annotation's schema
+    """
+    df = pandas.read_csv(path, index_col=indexCol).dropna(how='all')
+    df = df.dropna(axis=1, how='all')
+
+    return df
+
+
+def _query2df(syn, synId, clause):
+    """
+    Downloads the entity-view query result as a CsvFileTable then convert it into a pandas dataframe
```

Comment:
decided to go with entityview which has the same naming convention as the python client schema class type.